x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/math/MathOperationTests.java

```diff
@@ -35,4 +35,25 @@ public void testAbsPreservesType() {
         assertEquals(42f, MathOperation.ABS.apply(-42f));
         assertEquals(42d, MathOperation.ABS.apply(-42d));
     }
+
+    public void testSignIntegerType() {
+        assertEquals(-1, MathOperation.SIGN.apply((byte) -42));
+        assertEquals( -1, MathOperation.SIGN.apply((short) -42));
+        assertEquals(-1, MathOperation.SIGN.apply(-42));
+        assertEquals( -1, MathOperation.SIGN.apply((long) -42));
+        assertEquals(-1, MathOperation.SIGN.apply(-42.0f));
+        assertEquals(-1, MathOperation.SIGN.apply(-42.0d));
+        assertEquals( 1, MathOperation.SIGN.apply((byte) 42));
+        assertEquals( 1, MathOperation.SIGN.apply((short) 42));
+        assertEquals(1, MathOperation.SIGN.apply(42));
+        assertEquals( 1, MathOperation.SIGN.apply((long) 42));
+        assertEquals(1, MathOperation.SIGN.apply(42.0f));
+        assertEquals(1, MathOperation.SIGN.apply(42.0d));
+        assertEquals(0, MathOperation.SIGN.apply((byte) 0));
+        assertEquals( 0, MathOperation.SIGN.apply((short) 0));
+        assertEquals(0, MathOperation.SIGN.apply(0));
+        assertEquals( 0, MathOperation.SIGN.apply((long) 0));
+        assertEquals(0, MathOperation.SIGN.apply(-0.0f));
+        assertEquals(0, MathOperation.SIGN.apply(-0.0d));
+    }
```

Comment:
Since the assertion values are the same, please put the values in an array and iterate over them:
```java
List<Number> negative = asList(-42, Long.value(-42), etc...)
for (Number number : negative) {
	Number result = MathOperations.SIGN.apply(number);
    assertEquals(Integer.class, result.getClass());
    assertEquals(-1, result);
}
...
```

Reply:
Has been changed to an iterative way. PTAL.